crmsh/ui_corosync.py

```diff
@@ -47,6 +50,221 @@ def _diff_nodes(args):
+            nodeid = next((x.nodeid for x in nodes if x.name == name), -1)
+            if nodeid == -1:
+                logger.error(f'Unknown node {name}.')
+            node_addresses[nodeid] = addr
+        if not args.nodes and not args.options:
+            logger.info("Nothing is updated.")
+        else:
+            try:
+                self.__save_changed_config(
+                    lm,
+                    lm.update_node_addr(args.linknumber, node_addresses),
+                    reload=not args.nodes,    # changes to node addresses are not hot reloadable
+                )
+            except corosync.LinkManager.DuplicatedNodeAddressException as e:
+                node1 = next(x.name for x in nodes if x.nodeid == e.node1)
+                node2 = next(x.name for x in nodes if x.nodeid == e.node2)
+                logger.error('Duplicated ip address %s: used for both %s and %s.', e.address, e.node1, e.node2)
+                return False
+
+    def do_add(self, context, *argv):
+        lm = self.__load_and_validate_links()
+        try:
+            args = LinkArgumentParser().parse(False, False, argv)
+        except LinkArgumentParser.SyntaxException as e:
+            logger.error('%s', str(e))
+            print('Usage: link add <node>=<addr> ... [options <option>=<value> ...] ', file=sys.stderr)
+            return False
+        self._validate_node_addresses(dict(args.nodes))
+        nodes = next(x for x in lm.links() if x is not None).nodes
+        node_addresses: dict[int, str] = dict()
+        for name, addr in args.nodes:
+            nodeid = next((x.nodeid for x in nodes if x.name == name), -1)
+            if nodeid == -1:
+                logger.error(f'Unknown node {name}.')
+            node_addresses[nodeid] = addr
+        try:
+            self.__save_changed_config(
+                lm,
+                lm.add_link(node_addresses, args.options),
+                reload=True,
+            )
+        except corosync.LinkManager.MissingNodesException as e:
+            for nodeid in e.nodeids:
+                name = next(x.name for x in nodes if x.nodeid == nodeid)
+                logger.error('The address of node %s is not specified.', name)
+            logger.error('The address of all nodes must be specified when adding a new link.')
+            return False
+        except corosync.LinkManager.DuplicatedNodeAddressException as e:
+            node1 = next(x.name for x in nodes if x.nodeid == e.node1)
+            node2 = next(x.name for x in nodes if x.nodeid == e.node2)
+            logger.error('Duplicated ip address %s=%s: already used for %s=%s.', e.node1, e.address, e.node2, e.address)
+            return False
+
+
+    @command.completer(completers.call(lambda: [
+        str(link.linknumber)
+        for link in corosync.LinkManager.load_config_file().links()
+        if link.linknumber != 0
+    ]))
+    def do_remove(self, context, linknumber: str):
+        if not linknumber.isdecimal():
```

Comment:
To support crm corosync link remove -1

```suggestion
        if not linknumber.isdecimal() and linknumber != "-1":
```